woltka/file.py

```diff
@@ -156,15 +156,16 @@ def id2file_from_dir(dir_, ext=None, ids=None):
     """
     res = {}
     for fname in listdir(dir_):
-        try:
-            id_ = file2stem(fname, ext)
-        except ValueError:
-            continue
-        if ids and id_ not in ids:
-            continue
-        if id_ in res:
-            raise ValueError(f'Ambiguous files for ID: "{id_}".')
-        res[id_] = fname
+        if isfile(join(dir_, fname)):
+            try:
+                id_ = file2stem(fname, ext)
+            except ValueError:
+                continue
+            if ids and id_ not in ids:
+                continue
+            if id_ in res:
+                raise ValueError(f'Ambiguous files for ID: "{id_}".')
```

Comment:
You may want to also raise an error if multiple files have the same filename, ignoring lower/upper-case. This varies by OS (some treat these as separate files, some treat them as the same file), so being strict about not allowing conflicts regardless of case will probably be useful in the long run.

Reply:
Good idea! I know in Windows filename case is ignored, and the concern you proposed does matter. In Linux and Mac filenames are case-sensitive. Woltka will perhaps run on Windows. I haven't tested, but since it does not depend on scikit-bio, I guess it will run. So keeping Windows-compatibility is good. However I am not exactly sure whether the Mac/Linux community favors case insensitivities. So allow me to not make changes for now, but create an issue (#32 ) to remind future developers of this question, and propose for a discussion on which strategy we want to adopt.

Reply:
Ah, got it. I agree that this isn't something worth worrying about at this stage -- now that I think about it, I guess we can assume that whatever `listdir(dir_)` gives you will be unique files, as Python sees it -- and this should work as expected regardless of OS (the behavior may be different but that shouldn't be a concern for Woltka?).

... So yeah this sounds fine to me :)

Reply:
I think you are right! Regardless of OS, this design won't run into problems. The only scenario as far as I can think which will break the program is that the user supplies a sample-to-filepath mapping file, which is generated in Windows but is used in a Mac. I think this scenario is extremely rare and the user should be responsible.